components/app_bar/app_bar_plugin_component.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useSelector} from 'react-redux';
+import classNames from 'classnames';
+
+import {Tooltip} from 'react-bootstrap';
+
+import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+
+import {getActivePluginId} from 'selectors/rhs';
+
+import {PluginComponent} from 'types/store/plugins';
+import Constants from 'utils/constants';
+
+import OverlayTrigger from 'components/overlay_trigger';
+
+type PluginComponentProps = {
+    component: PluginComponent;
+}
+
+const AppBarPluginComponent = (props: PluginComponentProps) => {
+    const {component} = props;
+
+    const channel = useSelector(getCurrentChannel);
+    const activePluginId = useSelector(getActivePluginId);
```

Comment:
I'm wondering if it's best practice to make these child components completely controlled, and driven by exclusively props rather than redux.

Reply:
I would argue that if the parent needed this information, pass it as a prop, if not, fetch from the state.